From bff3b3bf7f331a3fd3161e886ce0c709f413b2bb Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 16 Dec 2024 15:47:05 +0200
Subject: [PATCH] [FSW-10767] fs1: load phy fw from flash after cold boot

PCIE Phy FW should be loaded from flash to PCIE Sram only after cold
boot or manual reset. Other reset sources do not affect the PCIE Sram
memory content so copying the PHY FW again is a waste of time.

Change-Id: Iae7b05e271f7a582620184b890e6dd02c71ca659
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 14474bf..63f95dc 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -580,7 +580,10 @@
 
 void hl_init_pcie_phy_fw(void)
 {
-	hl_pcie_phy_fw_copy();
+	u32 rst_src = hl_rst_src_get();
+
+	if (rst_src == HL_COLD_RST || rst_src == HL_MANUAL_RST)
+		hl_pcie_phy_fw_copy();
 }
 
 static bool image_is_load_req(struct desc_flags flags, u32 rst_src)
